test/layouts/test_xmonad.py

```diff
@@ -255,8 +266,30 @@ def test_tall_margins(manager):
     assert_focused(manager, "one")
     assert_dimensions(manager, 4, 4, 392, 588)
 
+    manager.test_window("three")
+    assert_focused(manager, "three")
+    assert_dimensions(manager, 404, 4, 388, 288)
+
+    manager.c.layout.next()
+    assert_focused(manager, "two")
+    assert_dimensions(manager, 404, 300, 388, 292)
+
+    manager.c.next_layout()
+
+    # x = screen_width * 0.5 + margin[3]
+    # y = screen_height / 2
+    # width = screen_width * 0.5 - margin[1] - margin[3] - 2 * border
+    # height = screen_height / 2 - margin[0] - margin[2] - 2 * border
+    #            + min(margin[0], margin[2]) # Combine top margin
+    assert_focused(manager, "two")
+    assert_dimensions(manager, 408, 300, 384, 290)
+
+    manager.c.layout.previous()
+    assert_focused(manager, "three")
+    assert_dimensions(manager, 408, 2, 384, 288)
```

Comment:
`two` and `three` are still different heights (but are closer now). Is this a rounding issue?